src/types/input.ts

```diff
@@ -8,6 +8,104 @@ export interface LabelAttributes {
   help?: string
 }
 
+export interface InputProps {
+  /**
+   * To set the value of the input element without using `v-model`.
+   * @defaults ''
+   */
+  modelValue?: string | number
+
+  /**
+   * String to be used as the input label.
+   * @defaults ''
+   */
+  label?: string
+
+  /**
+   * KCheckbox has an instance of KLabel for supporting tooltip text.
+   * Use the labelAttributes prop to configure the KLabel's props.
+   * @defaults {}
+   */
+  labelAttributes?: LabelAttributes
+
+  /**
+   * String to be displayed as help text.
+   * @defaults ''
+   */
+  help?: string
+
+  /**
+   * Boolean to indicate whether the element is in an error state and should apply error styling.
+   * @defaults false
+   */
+  error?: boolean
+
+  /**
+   * String to be displayed as an error message if `error` prop is `true`.
+   * This prop will supersede the `help` prop if both have a value and `error` is `true`.
+   * @defaults false
+   */
+  errorMessage?: string
+
+  /**
+   * Use this prop to specify a character limit for the input.
+   * See the @char-limit-exceeded event for more details.
+   * @defaults null
+   */
+  characterLimit?: number
```

Comment:
Shall we remove the default value `null` for this one? Otherwise it should be `number | null` here.

Reply:
IMHO, the characterLimit default value should be `0`, when it is `0`, meaning that the limit is `Infinite` or maybe we should use `Number.Infinity` here instead. But since `null` has been used here for this matter, maybe it is safer to keep it this way.